examples/training/style_transfer/ada_in/training.py

```diff
@@ -0,0 +1,191 @@
+style_images = os.listdir(flags.STYLE_IMAGES)
+style_images = [os.path.join(flags.STYLE_IMAGES, path) for path in style_images]
+
+# split the style images in train, val and test
+total_style_images = len(style_images)
+train_style = style_images[: int(0.8 * total_style_images)]
+val_style = style_images[int(0.8 * total_style_images) : int(0.9 * total_style_images)]
+test_style = style_images[int(0.9 * total_style_images) :]
+
+
+def decode_and_resize(image_path):
+    """Decodes and resizes an image from the image file path.
+
+    Args:
+        image_path: The image file path.
+        size: The size of the image to be resized to.
+
+    Returns:
+        A resized image.
+    """
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, channels=3)
+    image = tf.image.convert_image_dtype(image, dtype="float32")
+    image = tf.image.resize(image, IMAGE_SIZE)
+    return image
+
+def extract_image_from_voc(element):
+    """Extracts image from the PascalVOC dataset.
+
+    Args:
+        element: A dictionary of data.
+        size: The size of the image to be resized to.
+
+    Returns:
+        A resized image.
+    """
+    image = element["image"]
+    image = tf.image.convert_image_dtype(image, dtype="float32")
+    image = tf.image.resize(image, IMAGE_SIZE)
+    return image
+
+#Build the style and content tf.data datasets.
+train_style_ds = (
+   tf.data.Dataset.from_tensor_slices(train_style)
+   .map(decode_and_resize, num_parallel_calls=AUTOTUNE)
+   .repeat()
+)
+train_content_ds = tfds.load("voc", split="train").map(extract_image_from_voc).repeat()
+
+val_style_ds = (
+    tf.data.Dataset.from_tensor_slices(val_style)
+    .map(decode_and_resize, num_parallel_calls=AUTOTUNE)
+    .repeat()
+)
+val_content_ds = (
+    tfds.load("voc", split="validation").map(extract_image_from_voc).repeat()
+)
+
+test_style_ds = (
+    tf.data.Dataset.from_tensor_slices(test_style)
+    .map(decode_and_resize, num_parallel_calls=AUTOTUNE)
+    .repeat()
+)
+test_content_ds = (
+    tfds.load("voc", split="test")
+    .map(extract_image_from_voc, num_parallel_calls=AUTOTUNE)
+    .repeat()
+)
+
+# Zipping the style and content datasets.
+train_ds = (
+    tf.data.Dataset.zip((train_style_ds, train_content_ds))
+    .shuffle(BATCH_SIZE * 2)
+    .apply(tf.data.experimental.dense_to_ragged_batch(batch_size=BATCH_SIZE))
+    .prefetch(AUTOTUNE)
+)
+
+val_ds = (
+    tf.data.Dataset.zip((val_style_ds, val_content_ds))
+    .shuffle(BATCH_SIZE * 2)
+    .apply(tf.data.experimental.dense_to_ragged_batch(batch_size=BATCH_SIZE))
+    .prefetch(AUTOTUNE)
+)
+
+test_ds = (
+    tf.data.Dataset.zip((test_style_ds, test_content_ds))
+    .shuffle(BATCH_SIZE * 2)
+    .apply(tf.data.experimental.dense_to_ragged_batch(batch_size=BATCH_SIZE))
+    .prefetch(AUTOTUNE)
+)
+
+trainer = AdaInTrainer(include_rescaling=True, image_size=IMAGE_SIZE)
+optimizer = keras.optimizers.Adam(learning_rate=1e-5)
```

Comment:
another place to look as far as the bad results go is here;  the learning rate might be too low or too high.